test/framework/src/main/java/org/opensearch/test/ParameterizedOpenSearchIntegTestCase.java

```diff
@@ -0,0 +1,43 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.test;
+
+import org.junit.After;
+import org.junit.Before;
+import org.opensearch.action.admin.cluster.settings.ClusterUpdateSettingsResponse;
+import org.opensearch.common.settings.Settings;
+
+/**
+ * Base class for running the tests with parameterization of the dynamic settings
+ * For any class that wants to use parameterization, use @ParametersFactory to generate
+ * different params for dynamic settings. Refer SearchCancellationIT for an example.
+ * Note: this doesn't work for the parameterization of feature flag settings.
+ */
+public abstract class ParameterizedOpenSearchIntegTestCase extends OpenSearchIntegTestCase {
+
+    protected final Settings dynamicSettings;
+
+    public ParameterizedOpenSearchIntegTestCase(Settings dynamicSettings) {
+        this.dynamicSettings = dynamicSettings;
+    }
+
+    @Before
+    public void beforeTests() {
+        ClusterUpdateSettingsResponse clusterUpdateSettingsResponse = client().admin()
```

Comment:
I think you can remove the local variable.

Reply:
done